### PR TITLE
Fix segmaker task for new oss provider

### PR DESCRIPTION
**Pull request by pankajastro**
Fix segmaker task for new oss provider (apache-airflow-providers-amazon==8.0.0)

### CODE REVIEW

File: astronomer/providers/amazon/aws/operators/sagemaker.py [hunk 1]
```diff
@@ -60,10 +60,17 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
         """
         self.preprocess_config()
         processing_job_name = self.config["ProcessingJobName"]
-        if self.hook.find_processing_job_by_name(processing_job_name):
-            raise AirflowException(
-                f"A SageMaker processing job with name {processing_job_name} already exists."
-            )
+        try:
+            if bool(self.hook.count_processing_jobs_by_name(processing_job_name)):
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:64]:
```suggestion
            if self.hook.count_processing_jobs_by_name(processing_job_name):
```

**Comment by pankajkoti** [comment]
@pankajastro I have copied back these changes in PR https://github.com/astronomer/astronomer-providers/pull/972

**Comment by pankajastro** [comment]
> @pankajastro I have copied back these changes in PR #972 so that we can verify there that the tests pass.

I feel, ideally, we have merged this into your branch or cherry-pick the commit.

**Comment by pankajastro** [comment]
closing it

**Pull request closed without merging**